HIP/hip-632.md

```diff
@@ -10,65 +10,118 @@ status: Accepted
+1. As a developer I want to be able to distinguish between Hedera Account ID and aliases within a smart contract so I can resolve address mappings in balance, permission or similar scenarios.
 2. As a developer I want to be able to confirm account authorization within a contract using simple `ECDSA`/`ED2519` key raw signatures as I would on Ethereum using `ECRECOVER`.
 3. As a developer I want to be able to confirm account authorization for `ECDSA` accounts within a contract without being limited to `ECRECOVER`.
 4. As a developer I want to be able to confirm account authorization for accounts within a contract using protobuf signature maps in a similar fashion as I would on Ethereum using `ECRECOVER`.
   
 ## Specification
 
-A new system contract with functions to support address (Hedera Account ID and virtual) mapping reconciliation and authorization is proposed.
+A new system contract with functions to support address (Hedera Account ID and EVM address alias) mapping and authorization is proposed.
+
+This will aid developers to support smart contract migration logic for cases where aliases are encountered.
+Additionally, EVM developers, who were limited to `ECRECOVER` authorization flows, will be able to expand
+authorization checks to other key types. 
 
-This will aid developers to support smart contract migration logic for cases where Hedera Account ID or virtual address values are encountered. Additionally, EVM developers who were limited to `ECRECOVER`  authorization flows will be able to expand authorization checks to other key types. 
+HAS methods here can use EVM-styled 20-byte addresses which are either Hedera account num aliases or EVM address 
+aliases. 
+
+* In this document the word _alias_, by itself, means either kind of alias.
+
+  (n.b.: See HIP-583 for definitions of Hedera account num alias and EVM address alias.  Hedera key aliases, defined there,
+are not supported by these methods.)
 
 | hash | signature | return | description |
 | --- | --- | --- | --- |
-|  | getVirtualAddresses(address) | address[] | returns an array of virtual addresses for a given Hedera Account ID  |
-|  | getHederaAddress(address) | address | returns the top level Hedera Account ID if applicable |
-|  | isVirtualAddress(address) | bool | true if valid virtual address, false if long-zero or non existing account |
-|  | isAuthorized(address, messageHash, signatureBlob) | bool | true if account is authorized to carry out transaction execution on account. Accepts protobuf key signature blobs. May be used for ECDSA, ED25519 and complex key flows |
-|  | isAuthorizedRaw(address, messageHash, signatureBlob) | bool | true if account is authorized to carry out transaction execution on account. Accepts single key raw signature blobs (ECDSA and ED25519). This provides similar logic to ECRECOVER. |
+| `0xdea3d081` | getEvmAddressAlias(address) | address | given a Hedera Account ID (account num alias) returns the EVM address alias, if present |
```

Comment:
started implementation and noticed that the return type should probably be `(int64 responseCode, address alias)` to be consistent with the other view functions.  The same for `getHederaAccountNumAlias` below

Reply:
done